File: java/src/Behavioral/Visitor/AbstractTitleInfo.java
package Behavioral.Visitor;


//AbstractTitleInfo.java - the abstract Visitee

public abstract class AbstractTitleInfo {  
   private String titleName;    
   public final void setTitleName(String titleNameIn) {
       this.titleName = titleNameIn;
   }
   public final String getTitleName() {
       return this.titleName;
   }
    
   public abstract void accept(TitleBlurbVisitor titleBlurbVisitor);
}

File: javascript/src/Behavioral/Visitor/AbstractTitleInfo.js
(function() {
  define([], function() {
    var AbstractTitleInfo;
    return AbstractTitleInfo = (function() {
      function AbstractTitleInfo() {}

      AbstractTitleInfo.prototype.titleName = null;

      AbstractTitleInfo.prototype.setTitleName = function(titleNameIn) {
        return this.titleName = titleNameIn;
      };

      AbstractTitleInfo.prototype.getTitleName = function() {
        return this.titleName;
      };

      AbstractTitleInfo.prototype.accept = function(titleBlurbVisitor) {
        throw "Unimplemented method, AbstractTitleInfo.accept";
      };

      return AbstractTitleInfo;

    })();
  });

}).call(this);
